### PR TITLE
Fallback to tokens API if local tokenizer fails

### PR DESCRIPTION
**Pull request by abdullahkady**

### CODE REVIEW

File: src/cohere/client.py [hunk 1]
```diff
@@ -207,11 +207,16 @@ def tokenize(
...
-            return TokenizeResponse(tokens=tokens, token_strings=[])
-        else:
-            return super().tokenize(text=text, model=model, request_options=request_options)
+            try:
+                tokens = asyncio.run(local_tokenizers.local_tokenize(self, text=text, model=model))
+                return TokenizeResponse(tokens=tokens, token_strings=[])
+            except Exception:
+                opts["additional_headers"] = opts.get("additional_headers", {})
+                opts["additional_headers"]["sdk-api-warning-message"] = "offline_tokenizer_failed"
+        print(opts)
```
Comment (billytrend-cohere) [inline src/cohere/client.py:218]:
```suggestion
```

Comment (abdullahkady) [reply src/cohere/client.py:218]: woops

File: src/cohere/client.py [hunk 2]
```diff
@@ -223,12 +227,17 @@ def detokenize(
...
         # and the request will be processed using the offline tokenizer. If set to False, the request will be processed using the API. The default value is True.
+        opts: RequestOptions = request_options or {}  # type: ignore
         if offline:
-            model = model or "command"
-            text = asyncio.run(local_tokenizers.local_detokenize(self, model=model, tokens=tokens))
-            return DetokenizeResponse(text=text)
-        else:
-            return super().detokenize(tokens=tokens, model=model, request_options=request_options)
+            try:
+                model = model or "command"
```
Comment (abdullahkady) [inline src/cohere/client.py:233]:
```suggestion
```

**Review by billytrend-cohere — APPROVED** [review]
we can release in https://github.com/cohere-ai/cohere-python/pull/456

**Pull request merged**